python/export_util.py

```diff
@@ -128,6 +129,238 @@ def convert_to_isoformat(
+
+    return nodes + relationships
+
+
+@mgp.read_proc
+def cypher_all(
+    ctx: mgp.ProcCtx,
+    path: str = "",
+    config: mgp.Map = {},
+) -> mgp.Record(path=str, data=str):
+    """Exports the graph in cypher with all the constraints, indexes and triggers.
+    Args:
+        context (mgp.ProcCtx): Reference to the context execution.
+        path (str): A path to the file where the query results will be exported. Defaults to an empty string.
+        config : mgp.Map
+            stream (bool) = False: Flag to export the graph data to a stream.
+            write_properties (bool) = True: Flag to keep node and relationship properties. By default set to true.
+            write_triggers (bool) = True: Flag to export graph triggers.
+            write_indexes (bool) = True: Flag to export indexes.
+            write_constraints (bool) = True: Flag to export constraints.
+    Returns:
+        path (str): A path to the file where the query results are exported. If path is not provided, the output will be an empty string.
+        data (str): A stream of query results in a cypher format.
+    Raises:
+        PermissionError: If you provided file path that you have no permissions to write at.
+        OSError: If the file can't be opened or written to.
+    """
+
+    cypher = []
+
+    memgraph = gqlalchemy.Memgraph()
+
+    if config.get("write_triggers", True):
+        triggers = memgraph.execute_and_fetch("SHOW TRIGGERS;")
+        for trigger in triggers:
+            cypher.append(
+                f"CREATE TRIGGER {trigger['trigger name']} ON {trigger['event type']} {trigger['phase']} EXECUTE {trigger['statement']};"
+            )
+        cypher.append("")
+
+    if config.get("write_indexes", True):
+        constraints = memgraph.execute_and_fetch("SHOW CONSTRAINT INFO;")
+        for constraint in constraints:
+            constraint_type = constraint["constraint type"]
+
+            if constraint_type == "exists":
+                cypher.append(
+                    f"CREATE CONSTRAINT ON (n:{constraint['label']}) ASSERT EXISTS (n.{constraint['properties']});"
+                )
+            elif constraint_type == "unique":
+                properties = (
+                    [constraint["properties"]]
+                    if isinstance(constraint["properties"], str)
+                    else list(constraint["properties"])
+                )
+                cypher.append(
+                    f"CREATE CONSTRAINT ON (n:{constraint['label']}) ASSERT {'n.' + ', n.'.join(properties)} IS UNIQUE;"
+                )
+            else:
+                raise ValueError("Unknown constraint type.")
+        cypher.append("")
+
+    if config.get("write_constraints", True):
+        indexes = memgraph.execute_and_fetch("SHOW INDEX INFO;")
+        for index in indexes:
+            index_type = index["index type"]
+            if index_type == "label":
+                cypher.append(f"CREATE INDEX ON :{index['label']};")
```

Comment:
Can you actually execute this one? I think @antepusic researched this some time ago, and there are queries which we can't execute because they require memgraph to not have any currently running queries. I think these are CREATE INDEX, and some others. Can you ping Ante to point you to that file where it is specified inside MAGE module?